### PR TITLE
LPS-194119 Copy Document Types when coping a file and folders on D&M

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
https://liferay.atlassian.net/browse/LPS-194119

How to test it:

1. create a Document Type on Global Site (it works the same with a Parent site)
2. Create a document with previous Document Type - file 1
3. Create a Folder on Global Site
4. Create a document with the Document Type inside the folder
5. Copy file 1 to Site
6. - the file copied has the document type of global site
7. Copy folder to Site
8. - the file inside of the folder has the document type of global site

Other test

1. create a Document Type on a Asset Library (it works the same with a Parent site)
2. Create a document with previous Document Type - file 1
3. Create a Folder on Asset Library
4. Create a document with the Document Type inside the folder
5. Copy file 1 to Site
6. -  the file copied DOES NOT HAVE document type
7. Copy folder to Site
8. -  the file inside of the folder DOES NOT HAVE document type
9. connect site to asset library and make Structures available
10. delete the copied file and folder on Site (or copy on another folder)
11. Copy file 1 to Site
12. - the file copied has the document type
13. Copy folder to Site
14. - the file inside of the folder has the document type

- LPS-194119 add the file Type id to the copy of the file in the case that we have access to the structures from the source group
- LPS-194119 create a method to obtain the relation between file id and file type id on a treePath
- LPS-194119 add the map of the fileEntryTypeIds to the copy of the folder to add it on the copy of each file inside of the folder
- LPS-194119 add the fileEntryTypeId on the copy of the file entry
- LPS-194119 buildService
- LPS-194119 fix test

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLAppServiceImpl.java [hunk 1]
```diff
@@ -3332,13 +3342,20 @@ protected void copyFolderDependencies(
 
 				long[] assetCategoryIds = serviceContext.getAssetCategoryIds();
 				String[] assetTagNames = serviceContext.getAssetTagNames();
+				Long fileEntryTypeId = ParamUtil.getLong(
+					serviceContext, "fileEntryTypeId");
```
Comment (adolfopa) [inline portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLAppServiceImpl.java:3345]: Why isn't this a primitive `long`?

**Review by adolfopa — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:stop

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLAppServiceImpl.java (hunk 1 above)
Comment (AliciaGarciaGarcia) [reply portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLAppServiceImpl.java:3345]: that is what happens when you let intelliJ extract the variable :D

Thanks for pointing at it.

**Comment by AliciaGarciaGarcia** [comment]
change the log
rebased with master
and baselined again

no more changes done so I'll forward it

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:reevaluate:1317401_9224

**Comment by austinchiang** [comment]
Looks like this needs to be rebased again @AliciaGarciaGarcia

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
rebased

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Pull request closed without merging**